### ISSUE TITLE
Does label 0 means the fake image?

### ISSUE THREAD

**Issue by njuaplusplus** [RUB-SysSec/GANDCTAnalysis#11]
Hi,

Thank you for your paper and this repo. 

I have a question about the output of the pre-trained dct classifier. When I run the code as instructed in the readme:
```
python run_classifier.py submission_models/ffhq/ridge_dct ~/datasets/ffhq/fake -s 1000 -d mean_var/ffhq_mean_var
```

When the model output class 0, does it mean the image is fake?

Thanks,
Shengwei

**Comment by Joool** [comment]
Yeah, the labels are assigned here:

https://github.com/RUB-SysSec/GANDCTAnalysis/blob/master/prepare_dataset.py#L52-L54

Based on the ordered directory list, so fake will be given 0, real 1.

**Comment by njuaplusplus** [comment]
Thank you for your reply.

However, when I tested your pre-trained detector on 1k official FFHQ 1024x1024 real images, 998 were predicted as class 0.
When I tested it on 1k 1024x1024 fake images generated by stylegan2, 998 were predicted as class 0.

The command I used is
```
python run_classifier.py ridge_dct img_dir -s 1000 -d ffhq_mean_var/
```

Are there any details I missed?

**Comment by Joool** [comment]
Seems correct, please note that the classifier is trained on a dataset from whichfaceisreal.com not official FFHQ data.

**Comment by njuaplusplus** [comment]
Thank you for your reply.

Because in the beginning, I thought the artifacts came from the upsampling layers, and thus they should appear across different datasets.
Does this mean the detector kind of over-fit the training data instead of learning the general frequency features?

**Comment by Joool** [comment]
The point of the linear regression classifier was not to generalize across datasets. It was specifically designed to demonstrate that you can solve the dataset with a linear classifier (vs. more complicated methods like CNNs).

**Comment by njuaplusplus** [comment]
I see. Thank you.
